js/main.js

```diff
@@ -0,0 +1,91 @@
+var button = document.getElementById('buttonNews');
+button.onclick = function displayAndHide() {
+    var hide = document.getElementById('displayInfo');
+    if (hide.style.display === "") {
+        hide.style.display = 'none';
+    } else {
+        hide.style.display = '';
+    }
+}
+
+var submitBtn = document.querySelector('#submitBtn');
+submitBtn.addEventListener("click", checkName);
+
+function checkName(evt) {
+    evt.preventDefault();
+    var nameArea = document.querySelector("#nameArea");
+    if (nameArea.value == "") {
+        alert("Error: Name is empty!");
+        nameArea.focus();
+        return false;
+    }
+}
+
+submitBtn.addEventListener("click", checkEmail);
+
+function checkEmail(evt) {
+    evt.preventDefault();
+    var emailArea = document.querySelector("#emailArea");
+    var filter = /^([a-zA-Z0-9_\.\-])+\@(([a-zA-Z0-9\-])+\.)+([a-zA-Z0-9]{2,4})+$/;
+    if (emailArea.value == "") {
+        alert("Error: Email is empty!");
+        emailArea.focus();
+        return false;
+    }
+    if (!filter.test(emailArea.value)) {
+        alert('Please provide a valid email address');
+        emailArea.focus;
+        return false;
+    }
+}
+
+submitBtn.addEventListener("click", checkMobileNumber);
+function checkMobileNumber(number) {
+    number.preventDefault();
+    var phoneArea = document.querySelector("#phoneArea");
+    for (var i = 0; i < phoneArea.value.length; i++) {
+        if (phoneArea.value.length != 10 && phoneArea.value.length != 11) {
```

Comment:
this **if conditional** should not be inside the loop - you're now checking the same things for every character unnecessarily, you can just do the same check outside the loop once